src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -866,7 +879,21 @@ def _find_using_common_tenant(self, access_token, resource):
             subscriptions = self._find_using_specific_tenant(
                 tenant_id,
                 temp_credentials[_ACCESS_TOKEN])
-            all_subscriptions.extend(subscriptions)
```

Comment:
When are all_subscriptions different between old behavior and your new logic? when a subscription exists in multiple tenants? for example, it exists in two tenants, will it only show once in all_subscriptions?

Reply:
Yes. A subscription will only appear once since we are keying by `subscriptionId`. The old behavior is **newly listed ones overwriting previously listed ones**, while the new behavior is **newly listed ones being discarded**. Thus, we can make sure subscriptions from **user's** home tenant is preferred (home tenant is listed earlier).